### PR TITLE
Reader: New styles for full-post view, hook up back

### PR DESCRIPTION
**Pull request by blowery**
Add in styles for the new full post view. Scope some of these off the dynamic 'reader' group, which is set based on a feature flag.

Also hook up the back button and use the compact button for it.

Test live: https://calypso.live/?branch=add/reader-refresh/full-post-background

~~To test, start a local calypso like this: 
`make clean; ENABLE_FEATURES=reader/refresh-2016-07 make run`~~

I've enabled the feature flag, so it just works now

### CODE REVIEW

**Comment by blowery** [comment]
This doesn't have any of the breakpoint magic in it yet. Working on that.

File: client/components/reader-full-post/index.jsx [hunk 1]
```diff
@@ -21,19 +23,36 @@ import { fetchPost } from 'lib/feed-post-store/actions';
...
+				<StickyPanel className="reader-full-post__back-container">
+					<Button className="reader-full-post__back" borderless compact onClick={ this.handleBack }>
+						<Gridicon icon="arrow-left" /> { translate( 'Back' ) }
+					</Button>
 				</StickyPanel>
-				<ReaderFullPostHeader post={ this.props.post } />
-				<div dangerouslySetInnerHTML={ { __html: this.props.post.content } } />
-			</Main>
+				<div className="reader-full-post__content">
+					<div className="reader-full-post__sidebar">
```
Comment (bluefuton) [inline client/components/reader-full-post/index.jsx]: Can we drop this for the moment until author-compact-profile has landed? (#7127)

**Comment by bluefuton** [comment]
> This doesn't have any of the breakpoint magic in it yet. Working on that.

Will that be going in this PR or a shiny new one?

**Comment by blowery** [comment]
> Will that be going in this PR or a shiny new one?

Just pushed in the beginnings of it, and I think that's as far as I'm going to take it in this PR. Running into some issues where I'll have to refactor things afloat in other PRs.

File: client/components/reader-full-post/index.jsx (hunk 1 above)
Comment (blowery) [inline client/components/reader-full-post/index.jsx]: Let's just land #7127 first and I'll resolve the conflict after

**Comment by fraying** [comment]
Nice work, y'all!

Just a few notes (forgive me if this is premature):

- The "Reader" tab in the masterbar should be dark blue (it's "active" because you're in Reader).
- When the site is followed, the text "Following" should be the same green as the icon.
- The headline font size should be 36px (but I'm guessing the width hacking hasn't happened yet).
- When scrolling, the "back" link and the sidebar content should not move at all (they currently move a bit). 
- The fade applied to the tags should be like 20px wide. It's currently much wider.
- The follower count should be rounded (ie "3.7k followers" instead of "3756 followers"). (Separate PR?)


Various fiddling with whitespace:

fullpost-notes

- 20px between masterbar and "back"
- 60px between masterbar and gravatar
- 60px between masterbar and headline
- 22px between gravatar and author name
- 14px between author name and site title
- 22px between site title and follow
- 22px between follow and comment/like
- 22px between headline and meta info
- 32px between meta info and beginning of content

**Comment by blowery** [comment]
reader tab is now lit up. working through the rest.

**Comment by blowery** [comment]
> The headline font size should be 36px (but I'm guessing the width hacking hasn't happened yet).

Should it _always_ be 36px?

**Comment by fraying** [comment]
No. Just 36 in the big state we're looking at now. I believe I noted the sizes in the p2 post, but if I didn't I'll add it.

**Comment by blowery** [comment]
That would be handy. Any other font sizes that move around based on width?

**Comment by fraying** [comment]
Fullpost Font Sizes

Post Title
1280px: Size: 36px, Line-height: 46px
1040px: Size: 36px, Line-height: 46px
960px: Size: 36px, Line-height: 46px
660px: Size: 32px, Line-height: 40px
480px: Size: 26px, Line-height: 34px

Body Text
1280px: Size: 17px, Line-height: 28px
1040px: Size: 17px, Line-height: 28px
960px: Size: 17px, Line-height: 28px
660px: Size: 17px, Line-height: 28px
480px: Size: 15px, Line-height: 24px

**Comment by fraying** [comment]
Spacing issues on the comment/like links in the sidebar.

1. Comment icon should scoot 1px down, so that the bottom of the bubble is on the same baseline as the text.

screen shot 2016-08-11 at 2 06 06 pm

2. Comment and Like links should have about 30px of whitespace in between them

comm-like-space

**Comment by fraying** [comment]
name-site-colors

Looks like the author name and the site name are slightly different colors. They should be the same color. I think the top one is correct.

**Comment by fraying** [comment]
When the sidebar is present, the top of the headline should line up with the top of the gravatar, and the top of the content should line up with the bottom of the gravatar. Or as close as we can get it, at least.

Design:
screen shot 2016-08-11 at 2 21 38 pm

Current site:
screen shot 2016-08-11 at 2 23 48 pm

So it looks like, on the current site, the post title needs to move up 5px, and the content needs to move up 28px.

**Comment by fraying** [comment]
A note on text colors. Could we try these? (I'm limiting myself to the official colors.)

Sidebar (author name, site name, follower count, and comment/like counts):
Normal: Gray Lighten 10 (#a8bece)
Link: Medium Blue (#00aadc)
Hover: Light Blue (#78dcfa)

Date stamp and tags:
Normal: Gray Lighten 10 (#a8bece)
Link: Gray (#87a6bc)
Hover: Light Blue (#78dcfa)

Even though the link colors are different in the two sections (to denote importance), I'd like to try to make the normal text and hover colors consistent. Thoughts?

**Comment by fraying** [comment]
Tag fade is happening too soon. It should be about 45px to the right, fading out at the right edge of the content.

screen shot 2016-08-11 at 2 47 44 pm

**Comment by fraying** [comment]
Banish the RSS icon!

screen shot 2016-08-11 at 2 50 22 pm

Logic for the main avatar: Use the first one we find. All become circles.
- gravatar
- blavatar
- rss feed site image
- Apple touch icon from main site
- anything else we can think of
- a photo of one of my goats

If none of those are found, omit the avatar image entirely and slide everything up.

(Just kidding about the goats.)

**Comment by fraying** [comment]
ESC button. 

On the current site, the ESC keypress closes the fullpost view and returns the user to the main feed. We should keep this functionality in place for anyone who is used to it. Possible to add?

**Comment by fraying** [comment]
Reverse the order of comment and like in the sidebar.

Currently it's like and then comment. It should be comment and then like. 

(This is to match the order in the card view. Current site is share, comment, like, ellipsis.)

**Comment by fraying** [comment]
Comment Tweaks

Possible to implement these tweaks to comment display?

Design:
comments

Current Site:
screen shot 2016-08-11 at 3 11 50 pm
Example: https://calypso.live/read/feeds/2928910/posts/1117437071#comments

Changes:

• Add header
If 1: "1 COMMENT"
If more than one: "2 COMMENTS"
If 0, then "NO COMMENTS – Add the first!"
Standard refresh header CSS: SF UI Text, Medium, 14px, Gray Lighten 10 (#a8bece)

• Tweak to author display
Use same link colors as the post author.
If the commenter has a URL, pull in the site title. (Possible?)
Datestamp on 2nd line, 13px, Gray Lighten 20

• If the commenter has a URL, the gravatar should be linked, too.

• Remove extra space above Reply and Like.

• Remove "open in new window" icon. 

• I'd like the links on the commenter's name and site to open their site in Reader, if we're able. Is that possible?

**Comment by fraying** [comment]
I know there's a lot that goes into that first image space, but I'm wondering if we can remove the empty space on the left and right size in some consistent way? For example, this post: https://calypso.live/read/feeds/2928910/posts/1117437071#comments

screen shot 2016-08-11 at 3 21 00 pm

There's about 60px of space to the left and right of the image that isn't needed and kinda breaks the visual lines of the column. 

What I'd like to do is make sure the image uses the full width and crop from the top and bottom for images that are too large. Like so:

screen shot 2016-08-11 at 3 23 48 pm

Possible?

**Comment by fraying** [comment]
Problem with follower count / follow button wrapping.

When the follower count is sufficiently long, it wraps strangely. For example: https://calypso.live/read/feeds/27030/posts/1117210232

screen shot 2016-08-11 at 3 27 38 pm

When this element needs to wrap, it should do so thusly:

screen-shot-2016-08-11-fixed

**Comment by fraying** [comment]
Strange bug: occasionally I'm seeing unlinked author names.

For example: https://calypso.live/read/feeds/41437948/posts/1117086279

screen shot 2016-08-11 at 3 33 47 pm

**Comment by fraying** [comment]
Just a note that the new action buttons Daily Post implemented haven't made their way into this branch yet.

screen shot 2016-08-11 at 3 41 45 pm


Live Site: https://wordpress.com/read/feeds/27030/posts/1116987710

Staging: https://calypso.live/read/feeds/27030/posts/1116987710

**Comment by fraying** [comment]
I'd like to make a proposal. In situations where the Author Name and the Site Name are the same, like: https://calypso.live/read/feeds/24393283/posts/1116435392

screen shot 2016-08-11 at 3 44 35 pm

I'd like to suggest that we show the text only once, linking to the site view in Reader. This makes the display simpler, and keeps the user in Reader. Use the same bold treatment as the first one.

(In fact, I'd like all the site links to stay in Reader whenever possible.)

**Comment by fraying** [comment]
Ben asked me to post all of the comments above as Issues. So I did! https://github.com/Automattic/wp-calypso/milestone/110

**Comment by bluefuton** [comment]
Going to merge this and continue with fixes in subsequent PRs.

It's hiding behind a feature flag, so we should be good :)

File: client/reader/_style.scss [hunk 2]
```diff
@@ -1,3 +1,19 @@
+.is-group-reader-refresh .layout__content {
+ 	padding-top: 0;
```
Comment (aduth) [inline client/reader/_style.scss]: Appears this has unintended consequences on recommendations view:

screen shot 2016-09-26 at 8 27 21 pm

(Note lack of any padding above "We think you'll like", which differs from production)

**Pull request merged**